### PR TITLE
Replicate log message test failure

### PR DESCRIPTION
**Pull request by jonathan1055**
Issue #5572 

This first commit adds a test which fails at the latest dev of 12.x
It passes before the mr #5553 was committed.

### CODE REVIEW

File: sut/modules/unish/woot/drush.services.yml [hunk 1]
```diff
@@ -1,11 +1,16 @@
 services:
+  logger.channel.woot:
+    class: Drupal\Core\Logger\LoggerChannel
+    factory: logger.factory:get
```
Comment (greg-1-anderson) [inline sut/modules/unish/woot/drush.services.yml:4]: `drush.services.yml` is deprecated in Drush 12. A legacy service instantiator is provided for backwards compatibility purposes;  it is functional enough to satisfy the needs of most Drush commands, but it is not a full-service DI container. It does not support the `factory` element, for example.

I would suggest trying to set this up using the `create` method. See `WootStaticFactoryCommands` as an example.

**Comment by greg-1-anderson** [comment]
See LegacyServiceInstantiator for the code that emulates the Symfony DI container yml processing mechanism. Simple PRs that increase the compatibility would be welcome, but we don't want the adapter to get too complicated. Ideally, folks will port their command files instead.

**Comment by jonathan1055** [comment]
Thanks for the info. I have two questions though
(a) If I had not raised this issue, how would I know that `drush.services.yml` is deprecated in drush12? Maybe I missed some announcement? Usually when deprecated things are used we get messages. I just got a test failure, which was fortunate.
(b) Where would I find info on how to replace the functionality currently in `drush.services.yml`? I looked in `WootStaticFactoryCommands` for the example you gave, but it's not clear how I would go about making the changes

**Comment by weitzman** [comment]
a) We will be sure to put this into the Drush 12.0.0 release notes. I think a debug message noting use of the deprecated API would be useful. That would go somewhere in https://github.com/drush-ops/drush/blob/12.x/src/Runtime/LegacyServiceInstantiator.php
b) https://www.drush.org/12.x/commands/. Add a create() method and constructor to the commandfile. Remove drush.services.yml

**Comment by jonathan1055** [comment]
Thank you, that's really helpful.

**Pull request closed without merging**